src/Symfony/Bundle/FrameworkBundle/Resources/config/session.php

```diff
@@ -94,8 +100,8 @@
         ->set('session_listener', SessionListener::class)
             ->args([
                 service_locator([
-                    'session' => service('session')->ignoreOnInvalid(),
-                    'initialized_session' => service('session')->ignoreOnUninitialized(),
+                    'session' => service('.session.do-not-use')->ignoreOnInvalid(),
```

Comment:
well ... what's in the name :) it's gonna be removed in 6.0

id be tempted to make `.session` a non-shared factory service.

Reply:
you can't have 2 instances of Session at the same time.  It have to be shared

Reply:
shared from the request POV yes, what happens in DI is not necessarily related.

This effectively keeps a session service while the PR is about deprecating it :)

what' the proposal for 6.0 then? inline the session service arguments in the locator here, or keep the .session.do-not-use service?

Reply:
Well, in 6.0 we should delete all these services (state should be removed from DIC).
And we replace the SessionListener (that use internally the session service) by a SessionFactoryListener (that build the session.

Reply:
understood. we need to clean the way for 6.0 first, to then be able to refactor it more easy to a single factory approach.